realm/realm-library/src/main/cpp/jni_util/log.cpp

```diff
@@ -91,8 +92,9 @@ void JavaLogger::log(Log::Level level, const char* tag, jthrowable throwable, co
     // "JNI called with pending exception". This is something that should be avoided when printing log in JNI --
     // Always
     // print log before calling env->ThrowNew. Doing env->ExceptionCheck() here creates overhead for normal cases.
-    env->CallVoidMethod(m_java_logger, m_log_method, level, env->NewStringUTF(tag), throwable,
-                        env->NewStringUTF(message));
+    JavaLocalRef<jstring> java_tag(env, env->NewStringUTF(tag));
+    JavaLocalRef<jstring> java_error_message(env, env->NewStringUTF(message));
+    env->CallVoidMethod(m_java_logger, m_log_method, level, java_tag.get(), throwable, java_error_message.get());
```

Comment:
That sounds a like a good idea actually. As you point out, it is really easy to forget. I'll rather do that in another PR though.